src/socket.io/index.js

```diff
@@ -86,7 +86,13 @@ function onDisconnect(socket) {
 	plugins.hooks.fire('action:sockets.disconnect', { socket: socket });
 }
 
-function onConnect(socket) {
+async function onConnect(socket) {
+	try {
+		await validateSession(socket, false);
+	} catch (e) {
+		socket.emit('event:invalid_session');
```

Comment:
Should there be a `return;` here? If it's an invalid session no point adding them to rooms and triggering checkSession?

Reply:
good catch!